src/business-logic/services/test/execute-action-service.spec.ts

```diff
@@ -91,6 +91,18 @@ describe(ExecuteActionService.name, () => {
 
         expect(firstExecutedPiece.id).not.toBe(lastExecutedPiece.id)
       })
+
+      it('updates Piece ExecutedAt to be set', async () => {
+        const action: PieceAction = createPieceAction(PieceActionType.INSERT_PIECE_AS_ON_AIR)
+        const rundownServiceMock: RundownService = mock<RundownService>()
+
+        const testee: ExecuteActionService = createTestee({rundownService: rundownServiceMock}, {action})
+        await testee.executeAction(action.id, 'rundownId')
+
+        const [, executedPiece] = capture(rundownServiceMock.insertPieceAsOnAir).first()
+
+        expect(executedPiece.getExecutedAt()).not.toBe(undefined)
```

Comment:
You could also use the fake timers of jest to check that it is Date.now that is used for the executed at value.

Reply:
Not 100% sure that the change i did was the way you intended, but it works, and the test failes as it should, when the line being tested is missing.

Reply:
It was not what I intended :D Instead of manually overwriting `Date.now` which is now fixed to that mock for the remainder of the tests, you can use jests fake timers: https://jestjs.io/docs/timer-mocks

So you would, before the test, call `jest.useFakeTimers()` and after the test call `jest.useRealTimers()`. Remember that this should be outside the the scope of the test function. If it is done within the scope and the function throws, then the fake timers are leaking to other tests. You don't have to specify a Date, but just use the date that is locked by jest.

Reply:
Fixed